### PR TITLE
420 meta jen2

### PR DESCRIPTION
**Pull request by jenlampton**
another PR for https://github.com/backdrop/backdrop-issues/issues/420

### CODE REVIEW

File: core/modules/simpletest/tests/common.test [hunk 1]
```diff
@@ -2769,4 +2833,4 @@ class FeedIconTest extends BackdropWebTestCase {
     $this->assertEqual($matches[1], 'Subscribe to &amp;&quot;&#039;', 'theme_feed_icon() escapes reserved HTML characters.');
   }
 
-}
+}
```
Comment (quicksketch) [inline core/modules/simpletest/tests/common.test]: Oops, removed the ending line here. Should be restored.

File: core/includes/common.inc [hunk 2]
```diff
@@ -1159,13 +1160,45 @@ function backdrop_http_request($url, array $options = array()) {
...
+      $result->error = $result->status_message;
   }
 
   return $result;
 }
 
 /**
+ * Split an HTTP response status line into components.
+ *
+ * Refactored out of drupal_http_request() so it can be unit tested.
```
Comment (quicksketch) [inline core/includes/common.inc]: drupal > backdrop

**Comment by quicksketch** [comment]
This looks good. I'm a little perplexed by the replacement of a 1-liner in backdrop_http_request() to a 30-line function (plus tests). I read over the parent issue in https://www.drupal.org/node/205969 and I think it's an okay solution, although overly elaborate. I'd say let's keep this for consistency with D7, but it definitely looks like an overkill solution to the problem.

File: core/modules/simpletest/tests/common.test (hunk 1 above)
Comment (jenlampton) [inline core/modules/simpletest/tests/common.test]: restored

File: core/includes/common.inc (hunk 2 above)
Comment (jenlampton) [inline core/includes/common.inc]: fixed

**Comment by jenlampton** [comment]
pushed up new changes to PR.

**Comment by quicksketch** [comment]
Looks great. Thanks!

**Pull request merged**